chanzuckerberg/czgenepi
# Maya/184616 date range menu

### Summary
- **What:** Refactor of `DateFilter`
  - pull the menu out into its own component
  - pull out reusable functional code into util
  - pull constants out
  - pull types into their own file
- **Why:** This will facilitate the re-use of the filter logic for sample tree filtering
- **Ticket:** [[sc-184616]](https://app.shortcut.com/genepi/story/184616)
- **Env:** https://datefilter-frontend.dev.czgenepi.org/

### Testing
1. Ensure existing filters on the samples table still work
1. Ensure you can clear existing filters using chips

### Demos
No visual or functional changes.

### Checklist
